opsgenie/resource_opsgenie_incident_template_test.go

```diff
@@ -0,0 +1,167 @@
+
+var randomName = acctest.RandString(6)
+
+func init() {
+	resource.AddTestSweepers("opsgenie_incident_template", &resource.Sweeper{
+		Name: "opsgenie_incident_template",
+		F:    testSweepIncidentTemplate,
+	})
+}
+
+func testSweepIncidentTemplate(region string) error {
+	meta, err := sharedConfigForRegion()
+	if err != nil {
+		return err
+	}
+	client, err := incident.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+	if err != nil {
+		return err
+	}
+	if result != nil {
+		for _, value := range result.IncidentTemplates["incidentTemplates"] {
+			incidentName := "Incident Template Name-" + randomName
+			if strings.HasPrefix(value.Name, incidentName) {
+				log.Printf("Destroying incident template %s", value.Name)
+				deleteRequest := incident.DeleteIncidentTemplateRequest{IncidentTemplateId: value.IncidentTemplateId}
+				if _, err := client.DeleteIncidentTemplate(context.Background(), &deleteRequest); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func TestAccOpsGenieIncidentTemplate_basic(t *testing.T) {
+	config := testAccOpsGenieIncidentTemplate_basic(randomName)
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckOpsGenieIncidentTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOpsGenieIncidentTemplateExists(),
+				),
+			},
+		},
+	})
+}
+
+func testCheckOpsGenieIncidentTemplateDestroy(s *terraform.State) error {
+	client, err := incident.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opsgenie_incident_template" {
+			continue
+		}
+		result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+		if err != nil {
+			if result != nil {
+				for _, value := range result.IncidentTemplates["incidentTemplates"] {
+					incidentName := "Incident Template Name-" + randomName
+					if strings.HasPrefix(value.Name, incidentName) {
+						x := err.(*ogClient.ApiError)
+						if x.StatusCode != 404 {
+							return errors.New(fmt.Sprintf("Incident template still exists: %s", x.Error()))
+						}
+					}
+				}
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+func testCheckOpsGenieIncidentTemplateExists() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client, err := incident.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+		if err != nil {
+			return err
+		}
+		result, err := client.GetIncidentTemplate(context.Background(), &incident.GetIncidentTemplateRequest{})
+		if err != nil {
+			if result != nil {
+				for _, value := range result.IncidentTemplates["incidentTemplates"] {
+					incidentName := "Incident Template Name-" + randomName
```

Comment:
also here
we already have this in the tf config

Reply:
Changes done. removed randomName variable.